### ISSUE TITLE
option to fix intercept or slope in linear fit functions

### ISSUE THREAD

**Issue by jborbely** [MSLNZ/GTC#22]
For the line-fitting functions in `type_a`, namely,

`line_fit`
`line_fit_wls`
`line_fit_rwls`
`line_fit_wtls`

it could be useful to add the option to fix the intercept or slope to a specific value and allow the other parameter to vary for the fit.

For example, using the model `y=a*x`, by specifying `b=0`.

**Comment by BlairDHall** [comment]
Sure. Do you have a need for this? I'm not sure how useful it would actually be.
Another enhancement that might be popular is LS fitting to a more general function (e.g., quadratic, cubic, ...)

**Comment by jborbely** [comment]
I did have a need but since the option was not available in GTC I used scipy. I suppose the linear LS fitting implemented in GTC could be viewed as static and if someone wants something more customizable then they should perform the fit using different tools. Whether to implement a more general function could also be viewed similarly, other software packages exist and are designed for fitting so use them.
